# Estructuras y formato de los datos
Integrantes:
 - Fabricio Torres Samueza
 
 

## Ejercicio 1

### 1.a) Abrir el archivo

In [12]:
import pandas as pd

In [13]:
n_c =["age", "sex", "cp", "trestbps", "chol","fbs","restecg","thalach", "exang", "oldpeak", "slope", "ca", "thal" , "num"]
#df = pd.read_csv("processed.cleveland.data",  header=None )
df = pd.read_csv("processed.cleveland.data",  names=n_c )
df

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
298  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
299  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
300  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
301  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  num  
0      3.0  0.0  6.0    0  
1      2.0  3.0  3.0    2  
2      2.0  2.0  7.0    1  
3      3.0  0.0  3.0    0  
4      1.0  0.0  3.0    0  
..     ...  ...  ...  ...  
298    2.0  0.0  7.0    1  
299    2.0  2.0  7.0    2  
300    2.0  1.0  7.0    3  
301    2.0  1.0  3.0    1  
302    1.0    ?  3.0    0  

[303 rows x 14 columns]

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    float64
 1   sex       303 non-null    float64
 2   cp        303 non-null    float64
 3   trestbps  303 non-null    float64
 4   chol      303 non-null    float64
 5   fbs       303 non-null    float64
 6   restecg   303 non-null    float64
 7   thalach   303 non-null    float64
 8   exang     303 non-null    float64
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    float64
 11  ca        303 non-null    object 
 12  thal      303 non-null    object 
 13  num       303 non-null    int64  
dtypes: float64(11), int64(1), object(2)
memory usage: 33.3+ KB


 ###  1.b) Variables numéricas: 
También denominados variables estadísticas.

#### Discretas:
Son valores que están representados por números enteros.
- age
- trestbps 
- chol 
- thalach 
- ca

#### Continuas :
Pueden tomar cualquier valor en un rango.
 - oldepeak



### 1.c) Variables categóricas:
También denominados atributos.
#### Ordinales 
Una variable puede ser tratada como ordinal cuando sus valores representan categorías con alguna clasificación intrínseca.
- cp
- restcg 
- slope 
- thal

#### Nominal :
Una variable puede ser tratada como ordinal cuando sus valores representan categorías.
- sex
- exang
- num

### 1.d) Valores faltantes

En este análisis encontramos que tenemos algunos valores faltantes en la columna de 'ca' y 'thal'.

En este caso solo encontramos un solo tipo de símbolo '?'.

In [15]:
df.isnull().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
num         0
dtype: int64

In [16]:
df.ca.value_counts()

ca
0.0    176
1.0     65
2.0     38
3.0     20
?        4
Name: count, dtype: int64

In [17]:
df.ca.str.contains('?', regex=False).sum()

4

In [18]:
df.thal.value_counts()

thal
3.0    166
7.0    117
6.0     18
?        2
Name: count, dtype: int64

In [19]:
df.thal.str.contains('?', regex=False).sum()

2

### 1.e) Estudio de la variable "thal"
Como podemos observar la variable "thal" no se puede ordenar de mayor a menor ya que hay valores indeterminados, representados por "?".

Esta variable puede ser categótica ordinal ya que admite un orden jerarquico.


In [20]:
df.thal.value_counts()

thal
3.0    166
7.0    117
6.0     18
?        2
Name: count, dtype: int64

# Ejercicio 2

###  2.a) Carga del archivo

In [22]:
#leemos el archivo adult.data y cargamos el dataframe df_names
n_a = ["age", "workclass", "fnlwgt", "education", "education-num", "marital-status", "occupation" 
       ,"relationship", "race", "sex", "capital-gain", "capital-loss", "hours-per-week", "native-country" ,"income"]
df_names = pd.read_csv("adult.data.csv" , names=n_a)
df_names

age          workclass  fnlwgt    education  education-num  \
0       39          State-gov   77516    Bachelors             13   
1       50   Self-emp-not-inc   83311    Bachelors             13   
2       38            Private  215646      HS-grad              9   
3       53            Private  234721         11th              7   
4       28            Private  338409    Bachelors             13   
...    ...                ...     ...          ...            ...   
32556   27            Private  257302   Assoc-acdm             12   
32557   40            Private  154374      HS-grad              9   
32558   58            Private  151910      HS-grad              9   
32559   22            Private  201490      HS-grad              9   
32560   52       Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation    relationship    race  \
0            Never-married        Adm-clerical   Not-in-family   White   
1       Married-civ-spouse     Exec-managerial         Husband   White   
2                 Divorced   Handlers-cleaners   Not-in-family   White   
3       Married-civ-spouse   Handlers-cleaners         Husband   Black   
4       Married-civ-spouse      Prof-specialty            Wife   Black   
...                    ...                 ...             ...     ...   
32556   Married-civ-spouse        Tech-support            Wife   White   
32557   Married-civ-spouse   Machine-op-inspct         Husband   White   
32558              Widowed        Adm-clerical       Unmarried   White   
32559        Never-married        Adm-clerical       Own-child   White   
32560   Married-civ-spouse     Exec-managerial            Wife   White   

           sex  capital-gain  capital-loss  hours-per-week  native-country  \
0         Male          2174             0              40   United-States   
1         Male             0             0              13   United-States   
2         Male             0             0              40   United-States   
3         Male             0             0              40   United-States   
4       Female             0             0              40            Cuba   
...        ...           ...           ...             ...             ...   
32556   Female             0             0              38   United-States   
32557     Male             0             0              40   United-States   
32558   Female             0             0              40   United-States   
32559     Male             0             0              20   United-States   
32560   Female         15024             0              40   United-States   

       income  
0       <=50K  
1       <=50K  
2       <=50K  
3       <=50K  
4       <=50K  
...       ...  
32556   <=50K  
32557    >50K  
32558   <=50K  
32559   <=50K  
32560    >50K  

[32561 rows x 15 columns]

In [ ]:
df_names.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  income          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


Podemos observar que no existen valores nulos pero sí contiene valores 
indeterminados representados por "?".

Las variables que contienen valores indeterminados son:
- workclass
- occupation
- native-country

In [ ]:
df_names.workclass.str.contains('?', regex=False).sum()

1836

In [ ]:
df_names.education.str.contains('?', regex=False).sum()

0

In [ ]:
df_names["marital-status"].str.contains('?', regex=False).sum()

0

In [ ]:
df_names.occupation.str.contains('?', regex=False).sum()

1843

In [ ]:
df_names.relationship.str.contains('?', regex=False).sum()

0

In [ ]:
df_names.race.str.contains('?', regex=False).sum()

0

In [ ]:
df_names.sex.str.contains('?', regex=False).sum()

0

In [ ]:
df_names["native-country"].str.contains('?', regex=False).sum()

583

In [ ]:
df_names.income.str.contains('?', regex=False).sum()

0

### 2.b) Análisis de variables

#### Variables numéricas o cuantitativas
Obsevamos que existen 6 varibles numéricas tipo entero,podemos ver que todas las variables son discretas.

##### Discretas:
- age
- fnlwgt
- education-num
- capital-gain
- capital-loss
- hours-per-week


#### Variables categóricas o cualitativas
Podemos ver que de entre todas las varibles, 9 son variables categóricas-nominales ya que no admiten ninguna ordenación lógica.

##### Nominal:
- workclass
- marital-status
- education
- occupation 
- relationship  
- race     
- sex          
- native-countr
- income          

    


### 2.c) Elementos repetidos y únicos

Tenemos un total de 32561 filas y 15 columnas

In [ ]:
total = df_names.shape
print(total)

(32561, 15)


En nuestro DataFrame hay 24 registros repetidos

In [ ]:
repetidos= df_names.duplicated().sum()
print(repetidos)

24


Tenemos 32537 elementos únicos

In [ ]:
unicos = total[0] - repetidos
print(unicos)

32537


### 2.d) Análisis de elementos repetidos

Obtenemos un nuevo dataframe con los datos repetidos.

In [ ]:
df_dup = df_names[df_names.duplicated()]
df_dup

age          workclass  fnlwgt      education  education-num  \
4881    25            Private  308144      Bachelors             13   
5104    90            Private   52386   Some-college             10   
9171    21            Private  250051   Some-college             10   
11631   20            Private  107658   Some-college             10   
13084   25            Private  195994        1st-4th              2   
15059   21            Private  243368      Preschool              1   
17040   46            Private  173243        HS-grad              9   
18555   30            Private  144593        HS-grad              9   
18698   19            Private   97261        HS-grad              9   
21318   19            Private  138153   Some-college             10   
21490   19            Private  146679   Some-college             10   
21875   49            Private   31267        7th-8th              4   
22300   25            Private  195994        1st-4th              2   
22367   44            Private  367749      Bachelors             13   
22494   49   Self-emp-not-inc   43479   Some-college             10   
25872   23            Private  240137        5th-6th              3   
26313   28            Private  274679        Masters             14   
28230   27            Private  255582        HS-grad              9   
28522   42            Private  204235   Some-college             10   
28846   39            Private   30916        HS-grad              9   
29157   38            Private  207202        HS-grad              9   
30845   46            Private  133616   Some-college             10   
31993   19            Private  251579   Some-college             10   
32404   35            Private  379959        HS-grad              9   

            marital-status          occupation    relationship  \
4881         Never-married        Craft-repair   Not-in-family   
5104         Never-married       Other-service   Not-in-family   
9171         Never-married      Prof-specialty       Own-child   
11631        Never-married        Tech-support   Not-in-family   
13084        Never-married     Priv-house-serv   Not-in-family   
15059        Never-married     Farming-fishing   Not-in-family   
17040   Married-civ-spouse        Craft-repair         Husband   
18555        Never-married       Other-service   Not-in-family   
18698        Never-married     Farming-fishing   Not-in-family   
21318        Never-married        Adm-clerical       Own-child   
21490        Never-married     Exec-managerial       Own-child   
21875   Married-civ-spouse        Craft-repair         Husband   
22300        Never-married     Priv-house-serv   Not-in-family   
22367        Never-married      Prof-specialty   Not-in-family   
22494   Married-civ-spouse        Craft-repair         Husband   
25872        Never-married   Handlers-cleaners   Not-in-family   
26313        Never-married      Prof-specialty   Not-in-family   
28230        Never-married   Machine-op-inspct   Not-in-family   
28522   Married-civ-spouse      Prof-specialty         Husband   
28846   Married-civ-spouse        Craft-repair         Husband   
29157   Married-civ-spouse   Machine-op-inspct         Husband   
30845             Divorced        Adm-clerical       Unmarried   
31993        Never-married       Other-service       Own-child   
32404             Divorced       Other-service   Not-in-family   

                      race      sex  capital-gain  capital-loss  \
4881                 White     Male             0             0   
5104    Asian-Pac-Islander     Male             0             0   
9171                 White   Female             0             0   
11631                White   Female             0             0   
13084                White   Female             0             0   
15059                White     Male             0             0   
17040                White     Male             0             0   
18555                Black     Male         

Solamente uno de ellos tiene un país de origen indeterminado.

In [ ]:
df_dup["native-country"].value_counts()    

 United-States    17
 Mexico            4
 Guatemala         2
 ?                 1
Name: native-country, dtype: int64

Hay un 62.5% de hombres y un 37.5% de mujeres.

In [ ]:
df_dup.sex.value_counts(normalize=True)

 Male      0.625
 Female    0.375
Name: sex, dtype: float64

El 8.3% cobra más de 50 mil dólares.

In [ ]:
df_dup.income.value_counts(normalize=True)

 <=50K    0.916667
 >50K     0.083333
Name: income, dtype: float64

### 2.e) Análisis de elementos únicos

Obtenemos un nuevo dataframe con los datos únicos.

In [ ]:
df_uni = df_names.drop_duplicates()
df_uni

age          workclass  fnlwgt    education  education-num  \
0       39          State-gov   77516    Bachelors             13   
1       50   Self-emp-not-inc   83311    Bachelors             13   
2       38            Private  215646      HS-grad              9   
3       53            Private  234721         11th              7   
4       28            Private  338409    Bachelors             13   
...    ...                ...     ...          ...            ...   
32556   27            Private  257302   Assoc-acdm             12   
32557   40            Private  154374      HS-grad              9   
32558   58            Private  151910      HS-grad              9   
32559   22            Private  201490      HS-grad              9   
32560   52       Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation    relationship    race  \
0            Never-married        Adm-clerical   Not-in-family   White   
1       Married-civ-spouse     Exec-managerial         Husband   White   
2                 Divorced   Handlers-cleaners   Not-in-family   White   
3       Married-civ-spouse   Handlers-cleaners         Husband   Black   
4       Married-civ-spouse      Prof-specialty            Wife   Black   
...                    ...                 ...             ...     ...   
32556   Married-civ-spouse        Tech-support            Wife   White   
32557   Married-civ-spouse   Machine-op-inspct         Husband   White   
32558              Widowed        Adm-clerical       Unmarried   White   
32559        Never-married        Adm-clerical       Own-child   White   
32560   Married-civ-spouse     Exec-managerial            Wife   White   

           sex  capital-gain  capital-loss  hours-per-week  native-country  \
0         Male          2174             0              40   United-States   
1         Male             0             0              13   United-States   
2         Male             0             0              40   United-States   
3         Male             0             0              40   United-States   
4       Female             0             0              40            Cuba   
...        ...           ...           ...             ...             ...   
32556   Female             0             0              38   United-States   
32557     Male             0             0              40   United-States   
32558   Female             0             0              40   United-States   
32559     Male             0             0              20   United-States   
32560   Female         15024             0              40   United-States   

       income  
0       <=50K  
1       <=50K  
2       <=50K  
3       <=50K  
4       <=50K  
...       ...  
32556   <=50K  
32557    >50K  
32558   <=50K  
32559   <=50K  
32560    >50K  

[32537 rows x 15 columns]

Podemos observar que 582 tienen un país de origen indeterminado.

In [ ]:
df_new = df_uni["native-country"]
df_new.str.contains('?', regex=False).sum()

582

Podemos observar que 1836 tienen un tipo de trabajo indeterminado.

In [ ]:
df_uni.workclass.value_counts()
df_new2 = df_uni["workclass"]
df_new2.str.contains('?', regex=False).sum()

1836

Podemos observar que 27 tienen un país de origen y tipo de trabajo indeterminado.

In [ ]:
df_total = df_uni[(df_uni["native-country"].str.contains('?', regex=False) ) & ( df_uni["workclass"].str.contains('?', regex=False))]
df_total.shape

(27, 15)

In [ ]:
df_total

age workclass  fnlwgt      education  education-num  \
61      32         ?  293936        7th-8th              4   
297     39         ?  157443        Masters             14   
1152    24         ?   35633   Some-college             10   
1676    64         ?  168340        HS-grad              9   
2513    47         ?  174525        HS-grad              9   
3131    25         ?  237865   Some-college             10   
3579    21         ?  180303      Bachelors             13   
3834    32         ?  169886      Bachelors             13   
6059    35         ?  163582           10th              6   
7862    29         ?  125159   Some-college             10   
9616    32         ?  647882        HS-grad              9   
11614   35         ?  103710      Bachelors             13   
12996   37         ?  122265        HS-grad              9   
16488   33         ?  119918      Bachelors             13   
16838   28         ?  149646   Some-college             10   
18615   27         ?  251854      Bachelors             13   
20333   69         ?  199591    Prof-school             15   
20480   55         ?  316027        7th-8th              4   
23729   41         ?  211873      Assoc-voc             11   
23915   24         ?  311949        HS-grad              9   
25359   38         ?   94559      Bachelors             13   
26296   68         ?  286869        7th-8th              4   
26363   48         ?  353824   Some-college             10   
28194   27         ?  133359      Bachelors             13   
28688   20         ?  200061   Some-college             10   
30369   20         ?  250037   Some-college             10   
32525   81         ?  120478      Assoc-voc             11   

               marital-status occupation    relationship                 race  \
61      Married-spouse-absent          ?   Not-in-family                White   
297        Married-civ-spouse          ?            Wife   Asian-Pac-Islander   
1152            Never-married          ?   Not-in-family                White   
1676       Married-civ-spouse          ?         Husband                White   
2513       Married-civ-spouse          ?         Husband                White   
3131            Never-married          ?       Own-child                Black   
3579            Never-married          ?   Not-in-family   Asian-Pac-Islander   
3834            Never-married          ?   Not-in-family                White   
6059                 Divorced          ?       Unmarried                White   
7862            Never-married          ?   Not-in-family                Black   
9616       Married-civ-spouse          ?         Husband                White   
11614                Divorced          ?       Unmarried                White   
12996                Divorced          ?   Not-in-family   Asian-Pac-Islander   
16488           Never-married          ?   Not-in-family                Black   
16838                Divorced          ?       Own-child                White   
18615      Married-civ-spouse          ?            Wife                Black   
20333      Married-civ-spouse          ?            Wife                White   
20480      Married-civ-spouse          ?         Husband                White   
23729      Married-civ-spouse          ?            Wife                White   
23915           Never-married          ?   Not-in-family   Asian-Pac-Islander   
25359      Married-civ-spouse          ?            Wife                Other   
26296                 Widowed          ?   Not-in-family                White   
26363      Married-civ-spouse          ?         Husband                White   
28194   Married-spouse-absent          ?   Not-in-family                White   
28688           Never-married          ?       Own-child                White   
30369           Never-married          ?       Own-child                White   
32525                Divorced          ?       Unmarried                White   

 

Hay un 55.5% de mujeres y un 44.4% de hombres.

In [ ]:
df_total.sex.value_counts(normalize=True)

 Female    0.555556
 Male      0.444444
Name: sex, dtype: float64

El 85.1% cobra 50 mil dólares o menos y 14.8% que cobra más de 50 mil dólares.

In [ ]:
df_total.income.value_counts(normalize=True)

 <=50K    0.851852
 >50K     0.148148
Name: income, dtype: float64

Hay un total de 15 mujeres y 12 hombres.

De las 15 mujeres, 13 (86.6%) cobran 50 mil dólares o menos y 2 (13.3%) cobran más de 50 mil dólares.

De los 12 hombres, 10 (83.3%) cobran 50 mil dólares o menos y 2 (16.6%) cobran más de 50 mil dólares.

In [ ]:
df_sex_income = df_total[["sex", "income"]]
dfa = df_sex_income.groupby("sex").size()
dfa

sex
 Female    15
 Male      12
dtype: int64

In [ ]:
df_aux = df_sex_income.groupby(['sex', 'income']).size().reset_index(name='total')
df_aux

sex  income  total
0   Female   <=50K     13
1   Female    >50K      2
2     Male   <=50K     10
3     Male    >50K      2

In [ ]:
rest= df_aux.groupby("sex").sum()
f_value = rest.total[0]
m_value = rest.total[1]
df_aux["percent"] = df_aux.apply(lambda x:  x.total*100/f_value if "Female" in x.sex else  x.total*100/m_value , axis=1)
df_aux

sex  income  total    percent
0   Female   <=50K     13  86.666667
1   Female    >50K      2  13.333333
2     Male   <=50K     10  83.333333
3     Male    >50K      2  16.666667

### 2.f) Variable “fnlwgt”

En este caso la varible **fnlwgt** (peso final),  es el número de personas que el censo cree que representa la entrada.


Creo que sí tiene sentido eliminar los registros duplicados para no tener información repetida y poder realizar mejor el estudio ya que la presencia de datos duplicados puede tener un impacto en la toma de decisiones, la eficiencia operativa y también ocupan espacio innecesario en los serividores lo que conlleva un uso ineficiente de los recursos y un coste adicional, en nuestro ejemplo no son muchos y no tienen un efecto tan grande a la hora de realizar el estudio, pero en la vida real puede ser muy grande la cantidad de dato duplicados lo que sí acasionaría problemas.

In [ ]:
df_dup2 = df_names[df_names.duplicated(keep=False)]
df_dup2.sort_values(["fnlwgt"])

age          workclass  fnlwgt      education  education-num  \
28846   39            Private   30916        HS-grad              9   
25624   39            Private   30916        HS-grad              9   
21875   49            Private   31267        7th-8th              4   
7920    49            Private   31267        7th-8th              4   
22494   49   Self-emp-not-inc   43479   Some-college             10   
7053    49   Self-emp-not-inc   43479   Some-college             10   
2303    90            Private   52386   Some-college             10   
5104    90            Private   52386   Some-college             10   
18698   19            Private   97261        HS-grad              9   
17673   19            Private   97261        HS-grad              9   
5805    20            Private  107658   Some-college             10   
11631   20            Private  107658   Some-college             10   
11965   46            Private  133616   Some-college             10   
30845   46            Private  133616   Some-college             10   
6990    19            Private  138153   Some-college             10   
21318   19            Private  138153   Some-college             10   
16975   30            Private  144593        HS-grad              9   
18555   30            Private  144593        HS-grad              9   
15189   19            Private  146679   Some-college             10   
21490   19            Private  146679   Some-college             10   
16297   46            Private  173243        HS-grad              9   
17040   46            Private  173243        HS-grad              9   
5842    25            Private  195994        1st-4th              2   
13084   25            Private  195994        1st-4th              2   
22300   25            Private  195994        1st-4th              2   
28522   42            Private  204235   Some-college             10   
10367   42            Private  204235   Some-college             10   
4940    38            Private  207202        HS-grad              9   
29157   38            Private  207202        HS-grad              9   
21103   23            Private  240137        5th-6th              3   
25872   23            Private  240137        5th-6th              3   
8080    21            Private  243368      Preschool              1   
15059   21            Private  243368      Preschool              1   
4767    21            Private  250051   Some-college             10   
9171    21            Private  250051   Some-college             10   
31993   19            Private  251579   Some-college             10   
3917    19            Private  251579   Some-college             10   
28230   27            Private  255582        HS-grad              9   
5579    27            Private  255582        HS-grad              9   
26313   28            Private  274679        Masters             14   
8679    28            Private  274679        Masters             14   
4881    25            Private  308144      Bachelors             13   
4325    25            Private  308144      Bachelors             13   
17916   44            Private  367749      Bachelors             13   
22367   44            Private  367749      Bachelors             13   
16846   35            Private  379959        HS-grad              9   
32404   35            Private  379959        HS-grad              9   

            marital-status          occupation    relationship  \
28846   Married-civ-spouse        Craft-repair         Husband   
25624   Married-civ-spouse        Craft-repair         Husband   
21875   Married-civ-spouse        Craft-repair         Husband   
7920    Married-civ-spouse        Craft-repair         Husband   
22494   Married-civ-spouse        Craft-repair         Husband   
7053    Married-civ-spouse        Craft-repair         Husband   
2303         Never-married       Other-service   Not-in-family   
5104         Never-married       Other-service   Not-in-family   
1869

### 2.g) Dificultades encontradas

## Ejercicio 3

### 3.a) Análisis de variables

In [ ]:
df_accidente = pd.read_csv("accidentesBicicletas_2019.csv", delimiter=";")
df_accidente

num_expediente       fecha      hora  \
0      2019S000036  02/01/2019  20:45:00   
1      2019S000045  03/01/2019  10:30:00   
2      2019S000132  03/01/2019  12:45:00   
3      2019S000132  03/01/2019  12:45:00   
4      2019S000133  03/01/2019  14:30:00   
..             ...         ...       ...   
886    2019S039796  19/12/2019  11:40:00   
887    2019S039799  19/12/2019   9:45:00   
888    2019S039844  30/12/2019  12:05:00   
889    2019S039890  27/12/2019  18:30:00   
890    2019S039890  27/12/2019  18:30:00   

                                        localizacion numero  cod_distrito  \
0    AVDA. GRAN VIA DE HORTALEZA / GTA. LUIS ROSALES    65B            16   
1                        CTRA. DEHESA DE LA VILLA, 1      1             9   
2        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
3        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
4                           CALL. FELIPE ALVAREZ, 10     10            18   
..                                               ...    ...           ...   
886                          CALL. ANGEL GANIVET, 28     28             3   
887                CALL. WALIA / CALL. ANGEL GANIVET     21             3   
888                           CALL. PALAS DE REY, 27     27             8   
889                      CALL. ALEJANDRO SANCHEZ, 12     12            11   
890                      CALL. ALEJANDRO SANCHEZ, 12     12            11   

                distrito           tipo_accidente estado_meteorológico  \
0              HORTALEZA  Colisión fronto-lateral            Despejado   
1        MONCLOA-ARAVACA                    Caída            Despejado   
2      VILLA DE VALLECAS                  Alcance            Despejado   
3      VILLA DE VALLECAS                  Alcance            Despejado   
4      VILLA DE VALLECAS                  Alcance         Se desconoce   
..                   ...                      ...                  ...   
886               RETIRO                    Caída         Lluvia débil   
887               RETIRO                    Caída            Despejado   
888  FUENCARRAL-EL PARDO         Colisión lateral            Despejado   
889          CARABANCHEL      Atropello a persona         Se desconoce   
890          CARABANCHEL      Atropello a persona         Se desconoce   

                         tipo_vehículo tipo_persona       rango_edad    sexo  \
0                            Bicicleta    Conductor  De 70 a 74 años  Hombre   
1                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
2                            Bicicleta    Conductor  De 18 a 20 años  Hombre   
3                            Bicicleta    Conductor  De 21 a 24 años  Hombre   
4                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
..                                 ...          ...              ...     ...   
886                          Bicicleta    Conductor  De 55 a 59 años   Mujer   
887  Bicicleta EPAC (pedaleo asistido)    Conductor  De 40 a 44 años  Hombre   
888                          Bicicleta    Conductor  De 10 a 14 años   Mujer   
889                          Bicicleta    Conductor  De 30 a 34 años  Hombre   
890                          Bicicleta       Peatón  Menor de 5 años  Hombre   

     cod_lesividad                                     tipo_lesividad  \
0              3.0                        Ingreso superior a 24 horas   
1              2.0                Ingreso inferior o igual a 24 horas   
2              7.0  Asistencia sanitaria sólo en el lugar del acci...   
3             14.0                           Sin asistencia sanitaria   
4              7.0  Asistencia sanitaria sólo en el lugar del acci...   
..             ...                                                ...   
886            7.0  Asistencia sanitaria sólo en el lugar del acci...   
887            6.0  Asistencia sanitaria inmediata en centro de sa...   
888            7.0  Asistencia sanitaria só

#### Variables numéricas o cuantitativa

##### Discreta

- fecha
- cod_distrito
- rango_edad
- cod_lesividad
    
##### Continua

- hora
- coordenada_x_utm
- coordenada_y_utm


#### Variables categóricas o cualitativa

 ##### Nominal:

- num_expediente
- localizacion
- numero
- distrito
- tipo_accidente
- estado_meteorológico
- tipo_vehículo
- tipo_persona
- sexo
- positiva_alcohol
- positiva_droga
 ##### Ordinal:

- tipo_lesividad

In [ ]:
df_accidente.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 19 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   num_expediente        891 non-null    object 
 1   fecha                 891 non-null    object 
 2   hora                  891 non-null    object 
 3   localizacion          891 non-null    object 
 4   numero                891 non-null    object 
 5   cod_distrito          891 non-null    int64  
 6   distrito              891 non-null    object 
 7   tipo_accidente        891 non-null    object 
 8   estado_meteorológico  865 non-null    object 
 9   tipo_vehículo         891 non-null    object 
 10  tipo_persona          891 non-null    object 
 11  rango_edad            891 non-null    object 
 12  sexo                  891 non-null    object 
 13  cod_lesividad         795 non-null    float64
 14  tipo_lesividad        795 non-null    object 
 15  coordenada_x_utm      8

### 3.b) Valores faltantes

Podemos ver que hay valores faltantes en "estado_meteorológico", "cod_lesividad", "tipo_lesividad", "positiva_alcohol" y "positiva_droga"
También observamos que "sexo" y "rango_edad" tienen valores desconocidos.

In [ ]:
df_accidente.sexo.str.contains('Desconocido', regex=False).sum()

15

In [ ]:
df_accidente.rango_edad.str.contains('Desconocido', regex=False).sum()

16

#### 3.b.1) estado_meteorológico
Podemos observar que hay 26 valores faltantes.

In [ ]:
df_accidente.estado_meteorológico.isna().sum()

26

In [ ]:
df_estado_m = df_accidente[df_accidente.estado_meteorológico.isna()]
df_estado_m

num_expediente       fecha      hora  \
23     2019S003463  14/01/2019  22:45:00   
28     2019S004271  18/01/2019  18:05:00   
61     2019S009136  07/02/2019  16:10:00   
65     2019S009733  10/02/2019  16:05:00   
121    2019S015475  11/03/2019  21:30:00   
142    2019S016649  14/03/2019  11:55:00   
152    2019S017125  24/03/2019  21:45:00   
153    2019S017125  24/03/2019  21:45:00   
168    2019S017589  29/03/2019  20:50:00   
184    2019S018020  04/04/2019  21:40:00   
223    2019S019885  27/04/2019  13:40:00   
246    2019S020803  10/05/2019   9:50:00   
400    2019S024842  26/06/2019  17:06:00   
406    2019S025019  27/06/2019  20:15:00   
427    2019S025453  04/07/2019   1:20:00   
508    2019S027541  31/07/2019  22:00:00   
539    2019S028647  21/08/2019  19:45:00   
556    2019S029033  29/08/2019   2:35:00   
584    2019S029650  04/09/2019  15:50:00   
641    2019S031094  23/09/2019  18:10:00   
675    2019S031696  30/09/2019  15:30:00   
746    2019S033299  17/10/2019  15:30:00   
792    2019S035312  10/11/2019   0:10:00   
836    2019S037108  28/11/2019  19:30:00   
868    2019S038818  18/12/2019  15:00:00   
885    2019S039769  29/12/2019  15:30:00   

                                          localizacion numero  cod_distrito  \
23   CALL. MIGUEL ANGEL / PASEO. GENERAL MARTINEZ C...      9             7   
28                         AVDA. FERMINA SEVILLANO, 42     42            20   
61    CALL. VALLEHERMOSO / CALL. FERNANDEZ DE LOS RIOS     42             7   
65       PASEO. REINA CRISTINA / CALL. ANDRES TORREJON      9             3   
121                       PLAZA. ESPAÑA / CALL. BAILEN      8             1   
142                                 AVDA. VICTORIA, 11     11             9   
152                               CALL. HORTALEZA, 100    100             1   
153                               CALL. HORTALEZA, 100    100             1   
168                           AVDA. RAFAELA YBARRA, 73     73            12   
184            CALL. GRAN VIA / METRO. PLAZA DE ESPAÑA     80             1   
223                          AVDA. RAFAELA YBARRA, 137    137            17   
246                        CALL. ARROYO FONTARRON, 393    393            14   
400                           PASEO. INFANTA ISABEL, 4      4             3   
406                              CALL. TOMAS BRETON, 2      2             2   
427             CALL. CASINO / CALL. SANTIAGO EL VERDE      1             1   
508                       CALL. GENOVA / CALL. ZURBANO     15             7   
539       CALL. PRINCIPE DE VERGARA / CALL. HERMOSILLA     32             4   
556                     CALL. SEGOVIA / RONDA. SEGOVIA     51             2   
584    CALL. CAMINO VIEJO DE VILLAVERDE / CALL. UNIDAD      4            12   
641                          PLAZA. ALONSO MARTINEZ, 7      7             7   
675         CALL. DONOSO CORTES / CALL. ANDRES MELLADO     78             7   
746                                CALL. EVANGELIOS, 1      1            12   
792                            PLAZA. SAN ILDEFONSO, 1      1             1   
836                                 PASEO. CHOPERA, 10     10             2   
868                               GTA. RUIZ JIMENEZ, 2      2             7   
885           PASEO. ACACIAS / PLAZA. ORTEGA Y MUNILLA     49             2   

                distrito           tipo_accidente estado_meteorológico  \
23              CHAMBERÍ  Colisión fronto-lateral                  NaN   
28   SAN BLAS-CANILLEJAS         Colisión frontal                  NaN   
61              CHAMBERÍ                  Alcance                  NaN   
65                RETIRO  Colisión fronto-lateral                  NaN   
121               CENTRO                  Alcance                  NaN   
142      MONCLOA-ARAVACA  Colisión fronto-lateral                  NaN   
152               CENTRO                  Alcance                  NaN   
153               CENTRO                  Alcance                  NaN   
168            

#### 3.b.2) cod_lesividad
Podemos observar que hay 96 valores faltantes.

In [ ]:
df_accidente.cod_lesividad.isna().sum()

96

In [ ]:
df_estado_c_l = df_accidente[df_accidente.cod_lesividad.isna()]
df_estado_c_l

num_expediente       fecha      hora  \
9      2019S001678  09/01/2019  12:50:00   
23     2019S003463  14/01/2019  22:45:00   
28     2019S004271  18/01/2019  18:05:00   
44     2019S006649  30/01/2019  11:41:00   
56     2019S008627  06/02/2019  17:30:00   
..             ...         ...       ...   
854    2019S038240  11/12/2019  17:15:00   
868    2019S038818  18/12/2019  15:00:00   
883    2019S039726  15/11/2019  23:50:00   
884    2019S039749  28/12/2019   3:50:00   
885    2019S039769  29/12/2019  15:30:00   

                                          localizacion numero  cod_distrito  \
9                                   CALL. GRAN VIA, 66     66             1   
23   CALL. MIGUEL ANGEL / PASEO. GENERAL MARTINEZ C...      9             7   
28                         AVDA. FERMINA SEVILLANO, 42     42            20   
44                                CALL. TOMASA RUIZ, 1      1            11   
56                                   CALL. ALCALA, 173    173             4   
..                                                 ...    ...           ...   
854                         AVDA. LLANO CASTELLANO, 11     11             8   
868                               GTA. RUIZ JIMENEZ, 2      2             7   
883                  CALL. SAN MARCOS / CALL. BARBIERI     25             1   
884            CALL. GRAN VIA / CALL. DOCTOR CARRACIDO     67             1   
885           PASEO. ACACIAS / PLAZA. ORTEGA Y MUNILLA     49             2   

                distrito                tipo_accidente estado_meteorológico  \
9                 CENTRO                       Alcance            Despejado   
23              CHAMBERÍ       Colisión fronto-lateral                  NaN   
28   SAN BLAS-CANILLEJAS              Colisión frontal                  NaN   
44           CARABANCHEL  Choque contra obstáculo fijo            Despejado   
56             SALAMANCA                       Alcance            Despejado   
..                   ...                           ...                  ...   
854  FUENCARRAL-EL PARDO                       Alcance              Nublado   
868             CHAMBERÍ                          Otro                  NaN   
883               CENTRO       Colisión fronto-lateral            Despejado   
884               CENTRO                       Alcance            Despejado   
885           ARGANZUELA       Colisión fronto-lateral                  NaN   

    tipo_vehículo tipo_persona       rango_edad         sexo  cod_lesividad  \
9       Bicicleta    Conductor  De 15 a 17 años       Hombre            NaN   
23      Bicicleta    Conductor  De 25 a 29 años       Hombre            NaN   
28      Bicicleta    Conductor  De 45 a 49 años       Hombre            NaN   
44      Bicicleta    Conductor  De 30 a 34 años       Hombre            NaN   
56      Bicicleta    Conductor  De 40 a 44 años       Hombre            NaN   
..            ...          ...              ...          ...            ...   
854     Bicicleta    Conductor  De 35 a 39 años       Hombre            NaN   
868     Bicicleta    Conductor  De 25 a 29 años       Hombre            NaN   
883     Bicicleta    Conductor  De 35 a 39 años       Hombre            NaN   
884     Bicicleta    Conductor  De 18 a 20 años       Hombre            NaN   
885     Bicicleta    Conductor      Desconocido  Desconocido            NaN   

    tipo_lesividad coordenada_x_utm coordenada_y_utm positiva_alcohol  \
9              NaN      439.854.516    4.474.862.412                N   
23             NaN      441.368.335    4.476.287.483                N   
28             NaN      451.549.039    4.477.091.339                S   
44             NaN      439.683.831    4.471.991.241                N   
56             NaN      442.981.558    4.475.311.486                N   
..             ...              ...              ...              ...   
854            NaN      442.023.250    4.481.838.529                N   
868            NaN      440.114.247    4.475

#### 3.b.3) tipo_lesividad
Podemos observar que hay 96 valores faltantes.

In [ ]:
df_accidente.tipo_lesividad.isna().sum()

96

In [ ]:
df_estado_l = df_accidente[df_accidente.tipo_lesividad.isna()]
df_estado_l

num_expediente       fecha      hora  \
9      2019S001678  09/01/2019  12:50:00   
23     2019S003463  14/01/2019  22:45:00   
28     2019S004271  18/01/2019  18:05:00   
44     2019S006649  30/01/2019  11:41:00   
56     2019S008627  06/02/2019  17:30:00   
..             ...         ...       ...   
854    2019S038240  11/12/2019  17:15:00   
868    2019S038818  18/12/2019  15:00:00   
883    2019S039726  15/11/2019  23:50:00   
884    2019S039749  28/12/2019   3:50:00   
885    2019S039769  29/12/2019  15:30:00   

                                          localizacion numero  cod_distrito  \
9                                   CALL. GRAN VIA, 66     66             1   
23   CALL. MIGUEL ANGEL / PASEO. GENERAL MARTINEZ C...      9             7   
28                         AVDA. FERMINA SEVILLANO, 42     42            20   
44                                CALL. TOMASA RUIZ, 1      1            11   
56                                   CALL. ALCALA, 173    173             4   
..                                                 ...    ...           ...   
854                         AVDA. LLANO CASTELLANO, 11     11             8   
868                               GTA. RUIZ JIMENEZ, 2      2             7   
883                  CALL. SAN MARCOS / CALL. BARBIERI     25             1   
884            CALL. GRAN VIA / CALL. DOCTOR CARRACIDO     67             1   
885           PASEO. ACACIAS / PLAZA. ORTEGA Y MUNILLA     49             2   

                distrito                tipo_accidente estado_meteorológico  \
9                 CENTRO                       Alcance            Despejado   
23              CHAMBERÍ       Colisión fronto-lateral                  NaN   
28   SAN BLAS-CANILLEJAS              Colisión frontal                  NaN   
44           CARABANCHEL  Choque contra obstáculo fijo            Despejado   
56             SALAMANCA                       Alcance            Despejado   
..                   ...                           ...                  ...   
854  FUENCARRAL-EL PARDO                       Alcance              Nublado   
868             CHAMBERÍ                          Otro                  NaN   
883               CENTRO       Colisión fronto-lateral            Despejado   
884               CENTRO                       Alcance            Despejado   
885           ARGANZUELA       Colisión fronto-lateral                  NaN   

    tipo_vehículo tipo_persona       rango_edad         sexo  cod_lesividad  \
9       Bicicleta    Conductor  De 15 a 17 años       Hombre            NaN   
23      Bicicleta    Conductor  De 25 a 29 años       Hombre            NaN   
28      Bicicleta    Conductor  De 45 a 49 años       Hombre            NaN   
44      Bicicleta    Conductor  De 30 a 34 años       Hombre            NaN   
56      Bicicleta    Conductor  De 40 a 44 años       Hombre            NaN   
..            ...          ...              ...          ...            ...   
854     Bicicleta    Conductor  De 35 a 39 años       Hombre            NaN   
868     Bicicleta    Conductor  De 25 a 29 años       Hombre            NaN   
883     Bicicleta    Conductor  De 35 a 39 años       Hombre            NaN   
884     Bicicleta    Conductor  De 18 a 20 años       Hombre            NaN   
885     Bicicleta    Conductor      Desconocido  Desconocido            NaN   

    tipo_lesividad coordenada_x_utm coordenada_y_utm positiva_alcohol  \
9              NaN      439.854.516    4.474.862.412                N   
23             NaN      441.368.335    4.476.287.483                N   
28             NaN      451.549.039    4.477.091.339                S   
44             NaN      439.683.831    4.471.991.241                N   
56             NaN      442.981.558    4.475.311.486                N   
..             ...              ...              ...              ...   
854            NaN      442.023.250    4.481.838.529                N   
868            NaN      440.114.247    4.475

#### 3.b.4) positiva_alcohol
Podemos observar que hay 2 valores faltantes.

In [ ]:
df_accidente.positiva_alcohol.isna().sum()

2

In [ ]:
df_estado_p_a = df_accidente[df_accidente.positiva_alcohol.isna()]
df_estado_p_a

num_expediente       fecha      hora  \
427    2019S025453  04/07/2019   1:20:00   
756    2019S033667  22/10/2019  20:00:00   

                               localizacion numero  cod_distrito distrito  \
427  CALL. CASINO / CALL. SANTIAGO EL VERDE      1             1   CENTRO   
756                        PASEO. PRADO, 14     14             1   CENTRO   

       tipo_accidente estado_meteorológico tipo_vehículo tipo_persona  \
427  Colisión lateral                  NaN     Bicicleta    Conductor   
756           Alcance         Lluvia débil     Bicicleta    Conductor   

      rango_edad         sexo  cod_lesividad tipo_lesividad coordenada_x_utm  \
427  Desconocido  Desconocido            NaN            NaN      440.201.887   
756  Desconocido  Desconocido            NaN            NaN      441.109.260   

    coordenada_y_utm positiva_alcohol  positiva_droga  
427    4.473.158.988              NaN             NaN  
756    4.473.970.983              NaN             NaN

#### 3.b.5) positiva_droga
Podemos observar que hay 887 valores faltantes.

In [ ]:
df_accidente.positiva_droga.isna().sum()

887

In [ ]:
df_estado_a_p = df_accidente[df_accidente.positiva_droga.isna()]
df_estado_a_p

num_expediente       fecha      hora  \
0      2019S000036  02/01/2019  20:45:00   
1      2019S000045  03/01/2019  10:30:00   
2      2019S000132  03/01/2019  12:45:00   
3      2019S000132  03/01/2019  12:45:00   
4      2019S000133  03/01/2019  14:30:00   
..             ...         ...       ...   
886    2019S039796  19/12/2019  11:40:00   
887    2019S039799  19/12/2019   9:45:00   
888    2019S039844  30/12/2019  12:05:00   
889    2019S039890  27/12/2019  18:30:00   
890    2019S039890  27/12/2019  18:30:00   

                                        localizacion numero  cod_distrito  \
0    AVDA. GRAN VIA DE HORTALEZA / GTA. LUIS ROSALES    65B            16   
1                        CTRA. DEHESA DE LA VILLA, 1      1             9   
2        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
3        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA     64            18   
4                           CALL. FELIPE ALVAREZ, 10     10            18   
..                                               ...    ...           ...   
886                          CALL. ANGEL GANIVET, 28     28             3   
887                CALL. WALIA / CALL. ANGEL GANIVET     21             3   
888                           CALL. PALAS DE REY, 27     27             8   
889                      CALL. ALEJANDRO SANCHEZ, 12     12            11   
890                      CALL. ALEJANDRO SANCHEZ, 12     12            11   

                distrito           tipo_accidente estado_meteorológico  \
0              HORTALEZA  Colisión fronto-lateral            Despejado   
1        MONCLOA-ARAVACA                    Caída            Despejado   
2      VILLA DE VALLECAS                  Alcance            Despejado   
3      VILLA DE VALLECAS                  Alcance            Despejado   
4      VILLA DE VALLECAS                  Alcance         Se desconoce   
..                   ...                      ...                  ...   
886               RETIRO                    Caída         Lluvia débil   
887               RETIRO                    Caída            Despejado   
888  FUENCARRAL-EL PARDO         Colisión lateral            Despejado   
889          CARABANCHEL      Atropello a persona         Se desconoce   
890          CARABANCHEL      Atropello a persona         Se desconoce   

                         tipo_vehículo tipo_persona       rango_edad    sexo  \
0                            Bicicleta    Conductor  De 70 a 74 años  Hombre   
1                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
2                            Bicicleta    Conductor  De 18 a 20 años  Hombre   
3                            Bicicleta    Conductor  De 21 a 24 años  Hombre   
4                            Bicicleta    Conductor  De 15 a 17 años  Hombre   
..                                 ...          ...              ...     ...   
886                          Bicicleta    Conductor  De 55 a 59 años   Mujer   
887  Bicicleta EPAC (pedaleo asistido)    Conductor  De 40 a 44 años  Hombre   
888                          Bicicleta    Conductor  De 10 a 14 años   Mujer   
889                          Bicicleta    Conductor  De 30 a 34 años  Hombre   
890                          Bicicleta       Peatón  Menor de 5 años  Hombre   

     cod_lesividad                                     tipo_lesividad  \
0              3.0                        Ingreso superior a 24 horas   
1              2.0                Ingreso inferior o igual a 24 horas   
2              7.0  Asistencia sanitaria sólo en el lugar del acci...   
3             14.0                           Sin asistencia sanitaria   
4              7.0  Asistencia sanitaria sólo en el lugar del acci...   
..             ...                                                ...   
886            7.0  Asistencia sanitaria sólo en el lugar del acci...   
887            6.0  Asistencia sanitaria inmediata en centro de sa...   
888            7.0  Asistencia sanitaria só

#### 3.b.6) sexo
Podemos observar que hay 15 valores desconocidos.

In [ ]:
df_accidente.sexo.str.contains('Desconocido', regex=False).sum()

15

In [ ]:
df_sexo = df_accidente[df_accidente.sexo.str.contains('Desconocido', regex=False)]
df_sexo

num_expediente       fecha      hora  \
315    2019S022596  31/05/2019  17:05:00   
360    2019S023732  13/06/2019  21:35:00   
381    2019S024294  19/06/2019  10:30:00   
386    2019S024417  20/06/2019  17:27:00   
420    2019S025299  28/06/2019  23:00:00   
427    2019S025453  04/07/2019   1:20:00   
504    2019S027484  17/07/2019  20:10:00   
519    2019S027834  05/08/2019  16:33:00   
558    2019S029101  29/08/2019  14:45:00   
648    2019S031116  23/09/2019  21:15:00   
653    2019S031237  26/09/2019   9:25:00   
756    2019S033667  22/10/2019  20:00:00   
771    2019S034153  27/10/2019  22:15:00   
872    2019S039131  21/12/2019  12:10:00   
885    2019S039769  29/12/2019  15:30:00   

                                          localizacion   numero  cod_distrito  \
315                          CALL. SANTA ENGRACIA, 101      101             7   
360                              CALL. JACOMETREZO, 15       15             1   
381  AVDA. MONASTERIO DE EL ESCORIAL / AVDA. SANTUA...        3             8   
386                            CALL. SAN TELESFORO, 13       13            15   
420                             CALL. MARCELO USERA, 1        1            12   
427             CALL. CASINO / CALL. SANTIAGO EL VERDE        1             1   
504                          PQUE. MADRID RIO, +00300E  +00300E            10   
519                          CALL. LOPEZ DE HOYOS, 100      100             5   
558                        PASEO. ALBERTO PALACIOS, 11       11            17   
648                     PQUE. MADRID RIO, FAROLA 26-10  +00320I             2   
653        AVDA. GENERAL PERON / CALL. JULIAN BESTEIRO       24             6   
756                                   PASEO. PRADO, 14       14             1   
771                           CALL. LOPEZ DE HOYOS, 66       66             5   
872                                CALL. VILLAAMIL, 27       27             6   
885           PASEO. ACACIAS / PLAZA. ORTEGA Y MUNILLA       49             2   

                distrito                tipo_accidente estado_meteorológico  \
315             CHAMBERÍ           Atropello a persona            Despejado   
360               CENTRO           Atropello a persona            Despejado   
381  FUENCARRAL-EL PARDO  Choque contra obstáculo fijo            Despejado   
386        CIUDAD LINEAL  Choque contra obstáculo fijo            Despejado   
420                USERA           Atropello a persona            Despejado   
427               CENTRO              Colisión lateral                  NaN   
504               LATINA           Atropello a persona            Despejado   
519            CHAMARTÍN                         Caída            Despejado   
558           VILLAVERDE           Atropello a persona            Despejado   
648           ARGANZUELA              Colisión frontal            Despejado   
653               TETUÁN       Colisión fronto-lateral            Despejado   
756               CENTRO                       Alcance         Lluvia débil   
771            CHAMARTÍN                         Caída            Despejado   
872               TETUÁN           Atropello a persona            Despejado   
885           ARGANZUELA       Colisión fronto-lateral                  NaN   

    tipo_vehículo tipo_persona   rango_edad         sexo  cod_lesividad  \
315     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
360     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
381     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
386     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
420     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
427     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
504     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
519     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
558     Bicicleta    Conductor  Desconocido  Desconocido      

#### 3.b.7) rango_edad
Podemos observar que hay 16 valores desconocidos.

In [ ]:
df_accidente.rango_edad.str.contains('Desconocido', regex=False).sum()

16

In [ ]:
df_rango = df_accidente[df_accidente.rango_edad.str.contains('Desconocido', regex=False)]
df_rango

num_expediente       fecha      hora  \
315    2019S022596  31/05/2019  17:05:00   
360    2019S023732  13/06/2019  21:35:00   
381    2019S024294  19/06/2019  10:30:00   
386    2019S024417  20/06/2019  17:27:00   
420    2019S025299  28/06/2019  23:00:00   
427    2019S025453  04/07/2019   1:20:00   
504    2019S027484  17/07/2019  20:10:00   
519    2019S027834  05/08/2019  16:33:00   
558    2019S029101  29/08/2019  14:45:00   
648    2019S031116  23/09/2019  21:15:00   
653    2019S031237  26/09/2019   9:25:00   
756    2019S033667  22/10/2019  20:00:00   
769    2019S034064  27/10/2019   0:38:00   
771    2019S034153  27/10/2019  22:15:00   
872    2019S039131  21/12/2019  12:10:00   
885    2019S039769  29/12/2019  15:30:00   

                                          localizacion   numero  cod_distrito  \
315                          CALL. SANTA ENGRACIA, 101      101             7   
360                              CALL. JACOMETREZO, 15       15             1   
381  AVDA. MONASTERIO DE EL ESCORIAL / AVDA. SANTUA...        3             8   
386                            CALL. SAN TELESFORO, 13       13            15   
420                             CALL. MARCELO USERA, 1        1            12   
427             CALL. CASINO / CALL. SANTIAGO EL VERDE        1             1   
504                          PQUE. MADRID RIO, +00300E  +00300E            10   
519                          CALL. LOPEZ DE HOYOS, 100      100             5   
558                        PASEO. ALBERTO PALACIOS, 11       11            17   
648                     PQUE. MADRID RIO, FAROLA 26-10  +00320I             2   
653        AVDA. GENERAL PERON / CALL. JULIAN BESTEIRO       24             6   
756                                   PASEO. PRADO, 14       14             1   
769             PLAZA. ALONSO MARTINEZ / CALL. SAGASTA        1             7   
771                           CALL. LOPEZ DE HOYOS, 66       66             5   
872                                CALL. VILLAAMIL, 27       27             6   
885           PASEO. ACACIAS / PLAZA. ORTEGA Y MUNILLA       49             2   

                distrito                tipo_accidente estado_meteorológico  \
315             CHAMBERÍ           Atropello a persona            Despejado   
360               CENTRO           Atropello a persona            Despejado   
381  FUENCARRAL-EL PARDO  Choque contra obstáculo fijo            Despejado   
386        CIUDAD LINEAL  Choque contra obstáculo fijo            Despejado   
420                USERA           Atropello a persona            Despejado   
427               CENTRO              Colisión lateral                  NaN   
504               LATINA           Atropello a persona            Despejado   
519            CHAMARTÍN                         Caída            Despejado   
558           VILLAVERDE           Atropello a persona            Despejado   
648           ARGANZUELA              Colisión frontal            Despejado   
653               TETUÁN       Colisión fronto-lateral            Despejado   
756               CENTRO                       Alcance         Lluvia débil   
769             CHAMBERÍ           Atropello a persona            Despejado   
771            CHAMARTÍN                         Caída            Despejado   
872               TETUÁN           Atropello a persona            Despejado   
885           ARGANZUELA       Colisión fronto-lateral                  NaN   

    tipo_vehículo tipo_persona   rango_edad         sexo  cod_lesividad  \
315     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
360     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
381     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
386     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
420     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
427     Bicicleta    Conductor  Desconocido  Desconocido            NaN   
504     

### 3.c) Valores distintos "lesividad"
La variable "lesividad" presenta 8 valores distintos.
Esta variable es una variable categórica ordinal ya que puede admitir una ordenación lógica

In [ ]:
valores_lesividad = df_accidente.tipo_lesividad.value_counts()
print(len(valores_lesividad))

8


In [ ]:
valores_lesividad

Asistencia sanitaria sólo en el lugar del accidente          317
Ingreso inferior o igual a 24 horas                          132
Sin asistencia sanitaria                                     109
Asistencia sanitaria inmediata en centro de salud o mutua     95
Atención en urgencias sin posterior ingreso                   80
Asistencia sanitaria ambulatoria con posterioridad            37
Ingreso superior a 24 horas                                   24
Fallecido 24 horas                                             1
Name: tipo_lesividad, dtype: int64

### 3.d) Estudio valores de "lesividad"
Mi respuesta es igual que la dada en el apartado anterior, es una variable categórica ordinal ya que puede admitir una ordenación lógica.


Asistencia sanitaria sólo en el lugar del accidente          
Ingreso inferior o igual a 24 horas                          
Sin asistencia sanitaria                                     
Asistencia sanitaria inmediata en centro de salud o mutua     
Atención en urgencias sin posterior ingreso                  
Asistencia sanitaria ambulatoria con posterioridad            
Ingreso superior a 24 horas                                   
Fallecido 24 horas


Se podría codificar de está manera:

##### Leve (1):

- Asistencia sanitaria sólo en el lugar del accidente 
- Ingreso inferior o igual a 24 horas 
- Sin asistencia sanitaria 
- Asistencia sanitaria inmediata en centro de salud o mutua 
- Atención en urgencias sin posterior ingreso
- Asistencia sanitaria ambulatoria con posterioridad

##### Grave (2):

- Ingreso superior a 24 horas

##### Fallecido (3):

- Fallecido 24 horas

### 3.e) Proporción de  accidentes

Podemos observar que un 98.7% presentan un test de alcohol negativo mientras que un 1.2% presenta un test de alcohol positivo.

In [ ]:
df_accidente.positiva_alcohol.value_counts(normalize=True)

N    0.987627
S    0.012373
Name: positiva_alcohol, dtype: float64